### DISCUSSION TITLE
How to correctly use /self-service/settings/api, I'm only want to update  password. What should I do?

### DISCUSSION THREAD

**Discussion by UniverseChen** [ory/kratos#2537] [Q&A] [answered]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

I don't know how to update password and not find any supports.

### Reproducing the bug

Ubuntu20.04
kratos v0.10.1
postgres 14

-------------------------

Request:
GET http://127.0.0.1:4435/self-service/settings/api HTTP/1.1
Authorization: session ZXtN3ssI5GDQE1erLkWApJJ81euQVag6

4435 is public port
ZXtN3ssI5GDQE1erLkWApJJ81euQVag6 is the lastest session in table sessions

-------------------------

Response:
{"error":{"id":"session_inactive","code":401,"status":"Unauthorized","reason":"This endpoint can only be accessed with a valid session. Please log in and try again.","message":"request does not have a valid authentication session"}}

So what is right request method?

### Version

0.10.1

**Comment by jonnyijapan** [comment, marked as answer]
I am using this via the Go SDK. Documentation is here:

https://github.com/ory/kratos-client-go/blob/master/docs/V0alpha2Api.md#InitializeSelfServiceSettingsFlowWithoutBrowser

Specifically I am using this to change password of user.

```
import (
	// other imports here

	kratosclient "github.com/ory/kratos-client-go"
)
func changePasswordUsingToken(newPassword string, sessionToken string) bool {
	// From here: https://www.ory.sh/docs/kratos/self-service/flows/user-settings#code-examples-for-nodejs-reactjs-go-
	success := false
	apiClient := getPublicApiClient()

	flow, _, err1 := apiClient.V0alpha2Api.InitializeSelfServiceSettingsFlowWithoutBrowserExecute(kratosclient.V0alpha2ApiApiInitializeSelfServiceSettingsFlowWithoutBrowserRequest{}.XSessionToken(sessionToken))
	if err1 != nil {
		log.Println("(token) Failed InitializeSelfServiceSettingsFlowWithoutBrowserExecute: ", err1)
	} else {
		ctx := context.Background()

		// Submit the form
		_, response, err := apiClient.V0alpha2Api.SubmitSelfServiceSettingsFlow(ctx).Flow(flow.Id).XSessionToken(sessionToken).SubmitSelfServiceSettingsFlowBody(
			kratosclient.SubmitSelfServiceSettingsFlowWithPasswordMethodBodyAsSubmitSelfServiceSettingsFlowBody(&kratosclient.SubmitSelfServiceSettingsFlowWithPasswordMethodBody{
				Method:   "password",
				Password: newPassword,
			}),
		).Execute()
		if err != nil {
			log.Println("Failed SubmitSelfServiceSettingsFlow: ", err)
			log.Println("response:", response)
		} else {
			success = true
		}
	}

	return success
}

func getPublicApiClient() *kratosclient.APIClient {
	configuration := kratosclient.NewConfiguration()
	configuration.Servers = []kratosclient.ServerConfiguration{
		{
			URL: getKratosPublicUrl(), //"http://host.docker.internal:4433", // Kratos Public API - when running Kratos container on "same machine/host/localhost", as this app
			// URL: "http://127.0.0.1:4434", // Kratos Public API
		},
	}
	apiClient := kratosclient.NewAPIClient(configuration)
	return apiClient
}

func getKratosPublicUrl() string {
	return "http://host.docker.internal:4433"
}
```

**Comment by vinckr** [comment]
Hello @UniverseChen 
How are you initializing the flow - can you try if it works with curl? Does the above snippet help understand how the flow should go?
